src/Service.Host/client/src/components/ChangeRequests/ChangeTypes/BoardChange.js

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import Grid from '@mui/material/Grid';
+import {
+    Dropdown,
+    InputField,
+    Loading,
+    Typeahead,
+    collectionSelectorHelpers,
+    itemSelectorHelpers
+} from '@linn-it/linn-form-components-library';
+import Button from '@mui/material/Button';
+import Typography from '@mui/material/Typography';
+import boardsActions from '../../../actions/boardsActions';
+import boardActions from '../../../actions/boardActions';
+import history from '../../../history';
+
+function BoardChange({ item, creating, handleFieldChange }) {
+    const dispatch = useDispatch();
+
+    const boardsSearchResults = useSelector(state =>
+        collectionSelectorHelpers.getSearchItems(state.boards)
+    ).map?.(b => ({
+        id: b.id,
+        name: b.boardCode,
+        boardCode: b.boardCode,
+        description: b.description
+    }));
+
+    const boardsSearchLoading = useSelector(state =>
+        collectionSelectorHelpers.getSearchLoading(state.boards)
+    );
+
+    const board = useSelector(reduxState => itemSelectorHelpers.getItem(reduxState.board));
+    const boardLoading = useSelector(reduxState =>
+        itemSelectorHelpers.getItemLoading(reduxState.board)
+    );
+
+    const handleBoardChange = selectedBoard => {
+        handleFieldChange('boardCode', selectedBoard.boardCode);
+        handleFieldChange('boardDescription', selectedBoard.description);
+        dispatch(boardActions.fetch(selectedBoard.boardCode));
+    };
+
+    const revisionsList = () => {
+        if (board?.layouts) {
+            return board.layouts
+                .map(l =>
+                    l.revisions.map(r => ({
+                        displayText: r.revisionCode,
+                        id: r.revisionCode,
+                        layoutSequence: r.layoutSequence,
+                        versionNumber: r.versionNumber
+                    }))
+                )
+                .flat()
+                .sort((a, b) => {
+                    if (a.layoutSequence < b.layoutSequence) {
+                        return -1;
+                    }
+                    if (a.layoutSequence > b.layoutSequence) {
+                        return 1;
+                    }
+                    if (a.versionNumber < b.versionNumber) {
+                        return -1;
+                    }
+                    if (a.versionNumber > b.versionNumber) {
+                        return 1;
+                    }
+                    return 0;
+                });
+        }
+        return [];
+    };
+
+    return (
+        <>
+            {creating ? (
+                <>
+                    <Grid item xs={4}>
+                        <Typeahead
```

Comment:
Could try the <Search /> component instead of the Typeahead - think we should slowly move towards deprecating Typeahead if we can since search is a little easier to use